### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by DmytroOrlov**

### CODE REVIEW

File: server/test/service/ScalaTestDynamicRunnerTest.scala [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package service
 
-import org.scalatest.FlatSpecLike
+import org.scalatest._
+import service.ScalaTestRunnerTest._
 
-class ScalaTestDynamicRunnerTest extends ScalaTestRunnerTest with FlatSpecLike {
```
Comment (novakov-alexey) [inline server/test/service/ScalaTestDynamicRunnerTest.scala]: why did you remove inheritance from ScalaTestRunnerTest?

Comment (DmytroOrlov) [inline server/test/service/ScalaTestDynamicRunnerTest.scala]: your tests run twice ane some more side effects

File: server/test/service/ScalaTestRunnerTest.scala [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package service
 
 import org.scalatest.{FlatSpec, Matchers, Suite}
+import service.ScalaTestRunnerTest._
 
 class ScalaTestRunnerTest extends FlatSpec with Matchers {
   behavior of "ScalaTestRunner"
-  val correctSolution = "class A { def sleepIn(weekday: Boolean, vacation: Boolean): Boolean = {!weekday || vacation}}"
```
Comment (novakov-alexey) [inline server/test/service/ScalaTestRunnerTest.scala]: What is the reason to have val correctSolution in companion object, but incorrect solution val in class members?

Comment (DmytroOrlov) [inline server/test/service/ScalaTestRunnerTest.scala]: just the result of fixing current test structure

File: server/test/service/ScalaTestDynamicRunnerTest.scala (hunk 1 above)
Comment (novakov-alexey) [inline server/test/service/ScalaTestDynamicRunnerTest.scala]: Not really. getReport was override and Dynamic test is supposed to have 5 test case. That means Dynamic test using the same test cases as static runner and this prove that both Dynamic and Static Runners are working equally. Now, having no confirmation that Dynamic is not falling on the same test cases as for Static Runner we lost ability to check it without such test. So test should be restored.

File: server/test/service/ScalaTestDynamicRunnerTest.scala [hunk 3]
```diff
@@ -44,13 +45,6 @@ class ScalaTestDynamicRunnerTest extends ScalaTestRunnerTest with FlatSpecLike {
           }""".stripMargin
 
 
-  override def getReport(solution: String) = {
```
Comment (novakov-alexey) [inline server/test/service/ScalaTestDynamicRunnerTest.scala]: getReport is using another execSuite

File: server/test/service/ScalaTestRunnerTest.scala (hunk 2 above)
Comment (novakov-alexey) [inline server/test/service/ScalaTestRunnerTest.scala]: both are solution options and should placed closer to each other, otherwise I do not any reason to separate them

File: server/test/service/ScalaTestDynamicRunnerTest.scala (hunk 1 above)
Comment (novakov-alexey) [inline server/test/service/ScalaTestDynamicRunnerTest.scala]: btw, which side-effects do you mean?
